test/TritonGPU/amd/optimize-lds-usage.mlir

```diff
@@ -87,3 +87,28 @@ module attributes {"triton_gpu.num-warps" = 8 : i32, "triton_gpu.threads-per-war
     tt.return
   }
 }
+
+// -----
+
+// Check that optimization correctly handles LDS shortcuts
+// CHECK-LABEL: mfma_dot_shortcut
+// CHECK: %0 = triton_gpu.local_alloc %arg0 : (tensor<128x128xf16, #blocked>) -> !tt.memdesc<128x128xf16, #shared, #triton_gpu.shared_memory>
```

Comment:
Please use regex to replace those hardcoded `%0`, etc.

Reply:
Do you want me to change whole test file?

Reply:
Just this one for now--we can have another pull request to change others.